packages/video_player/video_player/lib/src/web_vtt.dart

```diff
@@ -0,0 +1,197 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:html/dom.dart';
+
+import 'closed_caption_file.dart';
+import 'package:html/parser.dart' as html_parser;
+
+/// Represents a [ClosedCaptionFile], parsed from the WebVtt file format.
+/// See: https://en.wikipedia.org/wiki/WebVtt
+class WebVttCaptionFile extends ClosedCaptionFile {
+  /// Parses a string into a [ClosedCaptionFile], assuming [fileContents] is in
+  /// the WebVtt file format.
+  /// * See: https://en.wikipedia.org/wiki/WebVtt
+  WebVttCaptionFile(this.fileContents)
+      : _captions = _parseCaptionsFromWebVttString(fileContents);
+
+  /// The entire body of the Vtt file.
+  final String fileContents;
+
+  @override
+  List<Caption> get captions => _captions;
+
+  final List<Caption> _captions;
+}
+
+List<Caption> _parseCaptionsFromWebVttString(String file) {
+  final List<Caption> captions = <Caption>[];
+
+  // Ignore metadata
+  List<String> metadata = ['HEADER', 'NOTE', 'REGION', 'WEBVTT'];
+
+  int captionNumber = 1;
+  for (List<String> captionLines in _readWebVttFile(file)) {
+    // CaptionLines represent a complete caption
+    // E.g
+    // [
+    //  [00:00.000 --> 01:24.000 align:center]
+    //  ['Introduction']
+    // ]
+    // if caption has just header or time, but no text, captionLines.length will be 1
+    if (captionLines.length < 2) continue;
+
+    // if caption has header equal metadata, ignore
+    String metadaType = captionLines[0]?.split(' ')[0];
+    if (metadata.contains(metadaType)) continue;
+
+    // Caption has header
+    bool hasHeader = captionLines.length > 2;
+    if (hasHeader && int.tryParse(captionLines[0]) != null) {
+      captionNumber = int.parse(captionLines[0]);
+    }
+
+    final _StartAndEnd startAndEnd = _StartAndEnd.fromWebVttString(
+      hasHeader ? captionLines[1] : captionLines[0],
+    );
+
+    final String text = captionLines.sublist(hasHeader ? 2 : 1).join('\n');
+
+    // TODO: Handle text formats
```

Comment:
Could you create an issue for this TODO? See TODO formats for flutter repos: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-checking-in-comments-that-ask-questions